### PR TITLE
main: move some duplicate code to a single place

### PR DESCRIPTION
**Pull request by aykevl**
This makes fixes like https://github.com/tinygo-org/tinygo/pull/915 easier to apply (and in fact fixes a similar issue with the mgba debugger support). And in general, avoiding duplication is a good thing.

I tested this on the GBA, microbit, pca10040, cortex-m-qemu, hifive1b, and host targets.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -363,66 +351,43 @@ func FlashGDB(pkgName string, ocdOutput bool, options *compileopts.Options) erro
...
 			gdbCommands = append(gdbCommands, "target remote :2345")
 
 			// Run in an emulator.
 			args := append(config.Target.Emulator[1:], tmppath, "-g")
-			daemon := exec.Command(config.Target.Emulator[0], args...)
+			daemon = exec.Command(config.Target.Emulator[0], args...)
 			daemon.Stdout = os.Stdout
 			daemon.Stderr = os.Stderr
+		case "msd":
+			return errors.New("gdb is not supported for drag-and-drop programmable devices")
```
Comment (deadprogram) [inline main.go]: How can you run GDB on devices such as ItsyBitsy-M0 with the debug pins attached to JLink device?

Comment (aykevl) [reply main.go]: By adding the `openocd-interface` and `openocd-target` keys to the JSON file (or I think with the `-programmer=jlink` flag). This PR doesn't change any of that, the change here is that some code got moved around.

Comment (deadprogram) [reply main.go]: Ah, OK. Thanks for the clarification.

**Comment by deadprogram** [comment]
Now merging.

**Pull request merged**